miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/ListTablesIT.java

```diff
@@ -0,0 +1,260 @@
+  }
+
+  @Test
+  public void testListBoxesSetup() throws Exception {
+    testTabbedPageSetup("boxes", boxesColumns);
+  }
+
+  @Test
+  public void testListBoxesColumnSort() throws Exception {
+    testTabbedColumnsSort("boxes");
+  }
+
+  // TODO: re-add once sequencers are listified
+  // @Test
+  // public void testListSequencersSetup() throws Exception {
+  // testTabbedPageSetup("sequencers", sequencersColumns);
+  // }
+  //
+  // @Test
+  // public void testListSequencersColumnSort() throws Exception {
+  // testTabbedColumnsSort("sequencers");
+  // }
+
+  @Test
+  public void testListKitsSetup() throws Exception {
+    testTabbedPageSetup("kitdescriptors", kitsColumns);
+  }
+
+  @Test
+  public void testListKitsColumnSort() throws Exception {
+    testTabbedColumnsSort("kitdescriptors");
+  }
+
+  @Test
+  public void testListIndicesSetup() throws Exception {
+    testTabbedPageSetup("indices", indicesColumns);
+  }
+
+  @Test
+  public void testListIndicesColumnSort() throws Exception {
+    testTabbedColumnsSort("indices");
+  }
+
+  @Test
+  public void testListStudiesSetup() throws Exception {
+    testPageSetup("studies", studiesColumns);
+  }
+
+  @Test
+  public void testListStudiesColumnSort() throws Exception {
+    testColumnsSort("studies");
+  }
+
+  @Test
+  public void testListPrintersSetup() throws Exception {
+    testPageSetup("printers", printersColumns);
+  }
+
+  @Test
+  public void testListPrintersColumnSort() throws Exception {
+    testColumnsSort("printers");
+  }
+
+  private void testPageSetup(String listTarget, Set<String> targetColumns) {
+    ListPage page = getList(listTarget);
+    DataTable table = page.getTable();
+    List<String> headings = table.getColumnHeadings();
+    assertEquals(targetColumns.size(), headings.size());
+    for (String col : targetColumns) {
+      assertTrue("Check for column: '" + col + "'", headings.contains(col));
+    }
+  }
+
+  private void testTabbedPageSetup(String listTarget, Set<String> targetColumns) {
+    ListTabbedPage page = getTabbedList(listTarget);
+    DataTable table = page.getSelectedTable();
+    List<String> headings = table.getColumnHeadings();
+    assertEquals(targetColumns.size(), headings.size());
+    for (String col : targetColumns) {
+      assertTrue("Check for column: '" + col + "'", headings.contains(col));
+    }
+  }
+
+  private void testColumnsSort(String listTarget) {
+    ListPage page = getList(listTarget);
+    DataTable table = page.getTable();
+    List<String> headings = table.getColumnHeadings();
+    headings.forEach(heading -> {
+      // sort one way
+      page.sortByColumn(heading);
+      assertTrue("sort once on column '" + heading + "' without errors", LimsUtils.isStringEmptyOrNull(page.getErrors().getText()));
+      // sort the other way
+      page.sortByColumn(heading);
+      assertTrue("sort twice on column '" + heading + "' without errors", LimsUtils.isStringEmptyOrNull(page.getErrors().getText()));
```

Comment:
Ideally this method would verify that the sorting worked rather than just make sure there's not an error. As long as there are two or more items, you could do something like

```
<sort once>
String ascRow1Val = <get value>, ascRow2Val <get value>
while (ascRow1Val.equals(ascRow2Val)) {
  <get row 2 and 3 (etc) instead>
}
<sort again and similarly determine descRow1Val and descRow2Val>
assertNotEquals(ascRow1Val.compareTo(ascRow2Val) > 0, descRow1Val.compareTo(descRow2Val) > 0);
```